src/Microsoft.Extensions.DependencyModel/DependencyContext.cs

```diff
@@ -68,48 +81,47 @@ public class DependencyContext
 
         public IReadOnlyList<RuntimeLibrary> RuntimeLibraries { get; }
 
-        public IReadOnlyList<KeyValuePair<string, string[]>> RuntimeGraph { get; }
+        public IReadOnlyList<RuntimeFallbacks> RuntimeGraph { get; }
+
+        public DependencyContext Merge(DependencyContext other)
+        {
+            if (other == null)
+            {
+                throw new ArgumentNullException(nameof(other));
+            }
+
+            return new DependencyContext(
+                TargetFramework,
+                Runtime,
+                IsPortable,
+                CompilationOptions,
+                CompileLibraries.Union(other.CompileLibraries, new LibraryMergeEqualityComparer<CompilationLibrary>()),
+                RuntimeLibraries.Union(other.RuntimeLibraries, new LibraryMergeEqualityComparer<RuntimeLibrary>()),
+                RuntimeGraph.Union(other.RuntimeGraph)
+                );
+        }
 
         private static DependencyContext LoadDefault()
         {
-            var entryAssembly = Assembly.GetEntryAssembly();
-            return Load(entryAssembly);
+            return DependencyContextLoader.Default.Load(Assembly.GetEntryAssembly());
         }
 
         public static DependencyContext Load(Assembly assembly)
         {
-            if (assembly == null)
-            {
-                throw new ArgumentNullException(nameof(assembly));
-            }
-
-            using (var stream = assembly.GetManifestResourceStream(assembly.GetName().Name + DepsJsonExtension))
-            {
-                if (stream != null)
-                {
-                    return new DependencyContextJsonReader().Read(stream);
-                }
-            }
+            return DependencyContextLoader.Default.Load(assembly);
+        }
 
-            var depsJsonFile = Path.ChangeExtension(assembly.Location, DepsJsonExtension);
-            if (File.Exists(depsJsonFile))
+        private class LibraryMergeEqualityComparer<T>: IEqualityComparer<T> where T:Library
+        {
+            public bool Equals(T x, T y)
             {
-                using (var stream = File.OpenRead(depsJsonFile))
-                {
-                    return new DependencyContextJsonReader().Read(stream);
-                }
+                return x.Name == y.Name;
```

Comment:
I also explicitly use `StringComparison.Ordinal` because I like typing, and am paranoid (as mentioned earlier). I'm not sure you need to be that paranoid ;P